### DISCUSSION TITLE
Cannot convert channel to list in DSL2

### DISCUSSION THREAD

**Discussion by jamesbaye** [nextflow-io/nextflow#2275] [Q&A] [answered]
### Expected behavior and actual behavior

Cannot convert channel to list in DSL2.

### Steps to reproduce the problem

```
ch = Channel.of(1,2,3).toList().val[0] # returns 1
```

```
nextflow.enable.dsl = 2
ch = Channel.of(1,2,3).toList().val[0] # hangs indefinitely
```

### Environment 

* Nextflow version: 21.04.3
* Java version: 1.8.0_282
* Operating system: CentOS Linux release 7.9.2009
* Bash version: 4.2.46(2)-release

### Additional context

We are interested in unit testing some internal channel creation methods, and would like to investigate channel contents as a list. Currently this bug is stopping us from writing those unit tests.

**Comment by pditommaso** [comment, marked as answer]
This is by design, you are expected to use the value returned by `toList`

**Reply by jamesbaye** [reply]
@pditommaso Do you mean by passing the channel as an input to a process? Is there any other way that can avoid the overhead of creating/queuing a process?

**Reply by pditommaso** [reply]
Use an operator eg [map](https://www.nextflow.io/docs/latest/operator.html?highlight=subscribe#map) or `subscribe`

**Reply by jamesbaye** [reply]
Thanks @pditommaso. Resolved, I didn't realise I could add assert statements inside `subscribe`. Was able to test everything from within subscribe.
